src/components/common/ContactPanelProvider/ContactPanelProvider.tsx

```diff
@@ -7,81 +7,46 @@ import {
+ * @param paramValue The of the contact id query param from the router
+ * @returns The prefix and contact id from the query param
  */
-export const getQueryContactId = (
-  query: ParsedUrlQuery,
-  contactIdParam: string,
-): string | undefined => {
-  const queryParam = query[contactIdParam];
-
-  if (!Array.isArray(queryParam)) {
-    return undefined;
-  }
-
-  // The contact id is the last item in the query param array, but it must be a UUID, not "map". See
-  // the comment in setQueryContactId for more details.
-  const contactId = queryParam.at(-1);
-  if (typeof contactId === 'string' && uuidRegex.test(contactId)) {
-    return contactId;
-  } else {
-    return undefined;
-  }
-};
-
-/**
- * Given a set of router query params, return new query params with the contact modified or removed.
- *
- * @param query The query params from the router
- * @param contactIdParam The name of the query param that holds the contact id (usually `'contactId'`)
- * @param contactId The new contact id (`undefined` closes the panel)
- * @returns The updated query params
- */
-export const setQueryContactId = (
-  query: ParsedUrlQuery,
-  contactIdParam: string,
-  contactId: string | undefined,
-): ParsedUrlQuery => {
+export const splitContactIdParam = (
+  paramValue: ParsedUrlQuery[string],
+): {
+  prefix: string[];
+  contactId: string | undefined;
+} => {
   // The contact id param is always an array because it corresponds to a router segment like
   // `[[...contactId]]`. The array can contain multiple elements, for example, on the contacts page
   // for routes like ".../contacts/map/:contactId". Regardless of the number of elements in the
   // array, if the contact id is present it is always the last element in the array. Note that the
-  // last element may not be the contact id if the contact id is not present: (e.g. ["map"] on the
+  // last element may not be the contact id if the contact id is not present: (e.g. ['map'] on the
   // contact map page). See the test cases for more examples.
 
-  // Clone the existing contact id array and remove the existing contact id if it is present
-  const contactIdValue = Array.isArray(query[contactIdParam])
-    ? [...query[contactIdParam]]
-    : [];
-  // The last element is probably the existing current contact id, but it could also be part of the
-  // route (e.g. `'flows'`)
-  const lastElement = contactIdValue.pop();
-  if (typeof lastElement === 'string' && !uuidRegex.test(lastElement)) {
-    // The last element is part of the route, not a contact id, so put it back
-    contactIdValue.push(lastElement);
+  if (!Array.isArray(paramValue)) {
+    return { prefix: [], contactId: undefined };
   }
 
-  // Now, add the new contact id
-  if (typeof contactId === 'string') {
-    contactIdValue.push(contactId);
+  // The contact id is the last item in the query param array, but it must be a UUID, not 'map', for
+  // example. The prefix is the rest of the param value, e.g. ['map'].
```

Comment:
What about personID, will this mess this up as it now gets added to the URL?

/list/contactUUID/personUUID